### PR TITLE
Improve ImportFromFileSystem script

### PR DESCRIPTION
**Pull request by aminelouati**

### CODE REVIEW

File: DataConnectorScripts/resources/catalog/ImportFromFileSystem.groovy [hunk 1]
```diff
@@ -159,14 +161,18 @@ void importFiles() {
         children = this.remoteFile.findFiles(new org.objectweb.proactive.extensions.dataspaces.vfs.selector.FileSelector(filePattern))
         children.each { f ->
             String relativePath = File.separator + remoteBasePath.getRelativeName(f.getName());
-            if (f.getType() == FileType.FILE) {
+            if (f.getType() == FileType.FILE && f.getContent().getSize() > 0 && !f.getContent().isOpen() && !f.isContentOpen()) {
```
Comment (fviale) [inline DataConnectorScripts/resources/catalog/ImportFromFileSystem.groovy:164]: can you explain the difference between f.getContent().isOpen() and f.isContentOpen() ?

File: DataConnectors/resources/catalog/FTP.xml [hunk 2]
```diff
@@ -30,6 +30,7 @@
         <variable name="FILE_PATTERN" value="file.txt" inherited="false" model="PA:NOT_EMPTY_STRING" description="Either a file name or a wildcard string pattern (e.g. file.txt, *.txt, **)" group="FTP Parameters"/>
         <variable name="REMOTE_BASE" value="" inherited="false" model="PA:REGEXP(^$|^.+/$)" description="The remote relative path on the FTP server from which we import file(s). $REMOTE_BASE can be either a path to a folder terminated by / or an empty value for the root (e.g. REMOTE/)" group="FTP Parameters"/>
         <variable name="LOCAL_BASE" value="" inherited="false" model="PA:REGEXP(^$|^.+/$)" description="Relative path of a folder inside the Global Data Space in which the downloaded data will be stored. LOCAL_BASE can be either a path to a folder terminated by / or an empty value for the root (e.g. LOCAL/, /global/path/LOCAL/)" group="FTP Parameters"/>
+        <variable name="CUT_FILE" value="false" inherited="false" model="PA:BOOLEAN" description="If true, delete imported file(s) in the remote SFTP server" group="FTP Parameters"/>
```
Comment (fviale) [inline DataConnectors/resources/catalog/FTP.xml:33]: as the number of variables in these workflows increase, shouldn't we add groups ?

File: DataConnectorScripts/resources/catalog/ImportFromFileSystem.groovy [hunk 3]
```diff
@@ -159,14 +161,18 @@ void importFiles() {
...
                 println("Examining remote file " + f.getName());
                 standardPath = new File(localDir, relativePath);
                 localUrl = standardPath.toURI().toURL();
                 println("Standard local path is " + standardPath);
                 LocalFile localFile = (LocalFile) fsManager.resolveFile(localUrl.toString());
                 println("Resolved local file name: " + localFile.getName());
                 createParentFolderAndCopyFile(localFile, f)
+                if(cut){
+                    println("Remote file "  + f.getName() + " is deleted")
```
Comment (fviale) [inline DataConnectorScripts/resources/catalog/ImportFromFileSystem.groovy:173]: `if (cut) {`

Also print should either be after the delete operation or before with:

println("Deleting remote file " + f.getName())

File: DataConnectors/resources/catalog/FTP.xml (hunk 2 above)
Comment (aminelouati) [reply DataConnectors/resources/catalog/FTP.xml:33]: We have already 2 groups: group="FTP Parameters" and group="FTP Connection"

File: DataConnectorScripts/resources/catalog/ImportFromFileSystem.groovy (hunk 3 above)
Comment (aminelouati) [reply DataConnectorScripts/resources/catalog/ImportFromFileSystem.groovy:173]: OK 👍

File: DataConnectorScripts/resources/catalog/ImportFromFileSystem.groovy (hunk 1 above)
Comment (fviale) [reply DataConnectorScripts/resources/catalog/ImportFromFileSystem.groovy:164]: one remark, does f.getContent().getSize() means that you are actually reading the file, thus influencing the result of isContentOpen() or isOpen()

Comment (fviale) [reply DataConnectorScripts/resources/catalog/ImportFromFileSystem.groovy:164]: Also, if the file is being written by another process, how would  f.getContent().getSize() behave ?

Comment (fviale) [reply DataConnectorScripts/resources/catalog/ImportFromFileSystem.groovy:164]: Possibly, to be on the safe side, you could do:

`f.getType() == FileType.FILE && !f.isContentOpen() &&  f.getContent().getSize() > 0`

**Pull request merged**